crates/cli/src/main.rs

```diff
@@ -219,46 +220,102 @@ fn main() -> io::Result<()> {
             Ok(0)
         }
         Some((CMD_FORMAT, matches)) => {
-            let maybe_values = matches.get_many::<OsString>(DIRECTORY_OR_FILES);
-
-            let mut values: Vec<OsString> = Vec::new();
-
-            match maybe_values {
-                None => {
-                    let mut os_string_values: Vec<OsString> = Vec::new();
-                    read_all_roc_files(
-                        &std::env::current_dir()?.as_os_str().to_os_string(),
-                        &mut os_string_values,
-                    )?;
-                    for os_string in os_string_values {
-                        values.push(os_string);
-                    }
-                }
-                Some(os_values) => {
-                    for os_string in os_values {
-                        values.push(os_string.to_owned());
-                    }
+            let from_stdin = matches.get_flag(FLAG_STDIN);
+            let to_stdout = matches.get_flag(FLAG_STDOUT);
+            let format_mode = if to_stdout {
+                FormatMode::WriteToStdout
+            } else {
+                match matches.get_flag(FLAG_CHECK) {
+                    true => FormatMode::CheckOnly,
+                    false => FormatMode::WriteToFile,
                 }
+            };
+
+            if from_stdin && matches!(format_mode, FormatMode::WriteToFile) {
+                eprintln!("When using the --stdin flag, either the --check or the --stdout flag must also be specified. (Otherwise, it's unclear what filename to write to!)");
+                std::process::exit(1);
             }
 
-            let mut roc_files = Vec::new();
+            let roc_files = {
+                let mut roc_files = Vec::new();
 
-            // Populate roc_files
-            for os_str in values {
-                let metadata = fs::metadata(os_str.clone())?;
-                roc_files_recursive(os_str.as_os_str(), metadata.file_type(), &mut roc_files)?;
-            }
+                let mut values: Vec<OsString> = Vec::new();
+
+                match matches.get_many::<OsString>(DIRECTORY_OR_FILES) {
+                    Some(os_values) => {
+                        for os_string in os_values {
+                            values.push(os_string.to_owned());
+                        }
```

Comment:
could be a `values.extend` right (also below)

Reply:
Nice, good point! Changed in 9e27145a52d66d263851465c5158e0441f52f92f